### PR TITLE
Refactor mssql migration

### PR DESCRIPTION
**Pull request by ssh24**
### Description
Refactor migration

#### Related issues
connect to https://github.com/strongloop/loopback-connector/issues/15

### CODE REVIEW

File: lib/migration.js [hunk 1]
```diff
@@ -66,60 +66,6 @@ function mixinMigration(MsSQL) {
...
-    }, cb);
-  };
-
-  MsSQL.prototype.isActual = function(cb) {
-    var ok = false;
-    var self = this;
-    async.each(Object.keys(this._models), function(model, done) {
-      self.showFields(model, function(err, fields) {
-        self.showIndexes(model, function(err, indexes) {
-          self.alterTable(model, fields, indexes, function(err, needAlter) {
```
Comment (jannyHou) [inline lib/migration.js]: If it calls `alterTable` here, we **could not** use the function in base connector.
`getAddModifyColumns` and `getColumnsToDrop` are missing.

File: lib/migration.js [hunk 2]
```diff
@@ -66,28 +66,6 @@ function mixinMigration(MsSQL) {
     });
   };
 
-  MsSQL.prototype.isActual = function(cb) {
-    var ok = false;
-    var self = this;
-    async.each(Object.keys(this._models), function(model, done) {
-      self.getTableStatus(model, function(err, fields, indexes) {
-        self.alterTable(model, fields, indexes, function(err, needAlter) {
```
Comment (jannyHou) [inline lib/migration.js]: Please keep `isActual` in mssql, since it's calling `alterTable`. We can improve it in the 2nd round.

Comment (ssh24) [inline lib/migration.js]: Fixed in commit https://github.com/strongloop/loopback-connector-mssql/pull/129/commits/002c2f081b7d4c0bab3a0b2bb85d4e9601648dec

**Comment by ssh24** [comment]
@slnode test please

**Comment by ssh24** [comment]
@slnode test please

**Comment by ssh24** [comment]
@slnode test please

**Pull request merged**